exercises/exercise_5/README.md

```diff
@@ -1,91 +1,160 @@
+- Disabling a proxy service because you already have a proxy in your cluster (yes, we don't have a reverse proxy, but you can add one in [Exercise 7](../exercise_7/README.md))
 
-Pull an application from a registry
+For this exercise, we are going to instrument only the `vote` and `results` services for disabling.
 
-Options:
-      --insecure   Use insecure registry, without SSL
-```
 
-* **`pull`** the image you just pushed, then pull the images of your neighbours
+## How to disable services
+
+As we learned in the previous exercise, we can add parameters to modify our compose file and allow overrides at deploy time. Since the compose file has to be rendered, the capability to disable a service was added.
+
+Starting with compose schema version ZZZ, any compose file can include extension fields, which all start with `x-`. If a service contains a `x-enabled` field, Docker App will only include that service if it's value is "truthy." For example...
 
-```sh
-$ docker-app pull myneighbourhublogin/hello:0.1.0
+```yaml
+services:
+  app:
+    x-enabled: false
+    image: nginx
 ```
 
-## inspect
+In this case, the `app` service will _always_ be disabled, as the value is false. However, we probably don't want to _always_ disable a service (why include it in the first place?). Therefore, it sounds like a great candidate for a parameter!
 
-`inspect` command you already know just work directly with a remote image stored  on a registry.
 
-* **`inspect`** your neighbours images
+## Updating the Vote and Results services
 
-```sh
-$ docker-app inspect myneighbour/hello:0.1.0
-words 0.1.0
+To update the vote and results services, let's do the following:
 
-Services (3) Replicas Ports Image
------------- -------- ----- -----
-web          1        33000 dockerdemos/lab-web
-words        3              dockerdemos/lab-words
-db           1              dockerdemos/lab-db
-```
+1. In the `vote` service, add a `x-enabled` field that has a value of `${vote.enabled}`.
 
-## install
+    <details>
+      <summary>Solution</summary>
 
-And of course the `install` command too works the same way!
+    ```yaml
+    services:
+      vote:
+        x-enabled: ${vote.enabled}
+    ```
+    </details>
 
-```sh
-$ docker-app install myneighbour/hello:0.1.0 --name app-pulled
-Creating network app-pulled_default
-Creating service app-pulled_web
-Creating service app-pulled_words
-Creating service app-pulled_db
-```
+2. Update the `parameters.yml` to set a default value for `vote.enabled` to `true`.
+
+    <details>
+      <summary>Solution</summary>
+
+    ```yaml
+    vote:
+      enabled: true
+    ```
+    </details>
+
+3. Do the same thing for the `results` service, but use `results.enabled` as the parameter name.
+
+4. Use the `docker app render` command to set the `vote.enabled` parameter to false. You'll see that the compose file no longer contains the `vote` service!
```

Comment:
Maybe using `docker app inspect -s vote.enabled=false` is simpler here? So it won't be shown in the services list?